src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/issueadjournment/IssueAdjournmentNoticeAboutToStartHandler.java

```diff
@@ -44,10 +45,14 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
 
         PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
 
-        if (sscsCaseData.getAdjournCasePreviewDocument() != null) {
+        // Give the user selects No to generate notice option and has uploaded their custom adjournment notice document.
+        // Then don't generate the system default adjournment notice document.
+        if (sscsCaseData.getAdjournCaseGenerateNotice().equalsIgnoreCase(YesNo.YES.getValue())) {
             previewService.preview(callback, DocumentType.ADJOURNMENT_NOTICE, userAuthorisation, true);
         } else {
-            response.addError("No draft adjournment notice found on case. Please use 'Adjourn case' event before trying to issue.");
+            if (sscsCaseData.getAdjournCasePreviewDocument() == null) {
```

Comment:
Do else if (...) Not else { if (...) { 
The latter is needlessly complex.
Using the nonNull(...) is also neater for the check

Reply:
Should there also be an error when ...generateNotice is Yes but ...preview document is null?

Reply:
There would not be an error if generateNotice is Yes. If Yes, then we don't care about the preview document and the system would generate the pre-defined template.